consensus/src/simplex/mod.rs

```diff
@@ -196,14 +197,73 @@ mod tests {
     use governor::Quota;
     use prometheus_client::registry::Registry;
     use prover::Prover;
-    use rand::Rng;
     use std::{
         collections::{BTreeMap, HashMap, HashSet},
         num::NonZeroU32,
         sync::{Arc, Mutex},
         time::Duration,
     };
-    use tracing::debug;
+
+    enum Action {
+        NewLink(Link),
+        Relink(Link),
+        Unlink,
+    }
+
+    enum Scope {
+        All,
+        Where(fn(usize, usize) -> bool),
+    }
+
+    fn only(x: usize, i: usize, j: usize) -> bool {
+        x == i || x == j
+    }
+    fn omit(x: usize, i: usize, j: usize) -> bool {
+        x != i && x != j
+    }
+
+    // Helper function to link validators together.
+    async fn link_validators(
+        oracle: &mut Oracle,
+        validators: &[Bytes],
+        action: Action,
+        scope: Scope,
+    ) {
+        for (i1, v1) in validators.iter().enumerate() {
+            for (i2, v2) in validators.iter().enumerate() {
+                // Ignore self
+                if v2 == v1 {
+                    continue;
+                }
+
+                // Scope only to certain connections
+                if let Scope::Where(f) = scope {
+                    if !f(i1, i2) {
+                        continue;
+                    }
+                }
+
+                // Do any unlinking first
+                match action {
+                    Action::Relink(_) | Action::Unlink => {
+                        oracle.remove_link(v1.clone(), v2.clone()).await.unwrap();
```

Comment:
I wonder if you want to drop the error here or not? I could imagine it to be useful to "clear all" (without having previously kept track of who was/wasn't linked).

Reply:
I could add something like a `force` or `force_remove` flag but I'll leave it as strict for now